### PR TITLE
Al update to use collection protocol

### PR DESCRIPTION
**Pull request by AlexHmelevskiAG**
Changes:
* updated the logic for Collection bin to use generic Collection protocol
* update the logic for xDiffCollection to use generic Collection protocol
* added constrained methods to xDiffCollection to use filtering
* use standard template for pod

### CODE REVIEW

**Comment by AlexHmelevskiAG** [comment]
Due to changing template of the pod it shows all the files as new :(

**Review by ghost — DISMISSED** [review]
lgtm

File: xDiffCollection/Classes/CollectionBin.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    
+    
+    public func updating(_ element: T,
+                         whereUnique unique: CollectionFilter<T>,
+                         whereCompare compare: CollectionFilter<T>) -> BinUpdateResult<T, CollectionBin> {
+        
+        let idx = self.index(where: unique)
+        
+        guard _filter(element) else {
+            return idx.map({ removingResult(for: $0, where: unique) })  ?? unchagedBinResut
```
Comment (piyush9290) [inline xDiffCollection/Classes/CollectionBin.swift]: Can you rename `unchangedBinResut` to `unchangedBinResult`

**Review by piyush9290 — APPROVED** [review]
👍

**Review by beljamali — APPROVED** [review]
👍

**Pull request merged**